operational_analysis/methods/plant_analysis.py

```diff
@@ -672,38 +786,72 @@ def run_AEP_monte_carlo(self):
 
         num_sim = self.num_sim
 
+        # Initialize arrays to store metrics and results
+        self._mc_num_points = np.empty(num_sim, dtype=np.float64)
+        self._r2_score = np.empty(num_sim, dtype=np.float64)
+        self._mse_score = np.empty(num_sim, dtype=np.float64)
+        
+        num_vars = 1
+        if self.reg_winddirection == 'Y' :
+            num_vars = num_vars + 2
+        if self.reg_temperature == 'Y':
+            num_vars = num_vars + 1
+                
+        if self.reg_model == 'lin':
+            self._mc_intercept = np.empty(num_sim, dtype=np.float64) 
+            self._mc_slope = np.empty([num_sim,num_vars], dtype=np.float64)
+
         aep_GWh = np.empty(num_sim)
         avail_pct =  np.empty(num_sim)
         curt_pct =  np.empty(num_sim)
         lt_por_ratio =  np.empty(num_sim)
 
         # Loop through number of simulations, run regression each time, store AEP results
         for n in tqdm(np.arange(num_sim)):
-            slope, intercept = self.run_regression(n)  # Get slope, intercept from regression
-            ws_lt = self.sample_long_term_reanalysis(self._mc_num_years_windiness[n],
-                                                     self._mc_reanalysis_product[n])  # Get long-term wind speeds
             
-            # Get long-term normalized gross energy by applying regression result to long-term monthly wind speeds
-            gross_norm_lt = ws_lt.multiply(slope) + intercept
-            gross_lt=gross_norm_lt*self.num_days_lt/30 # Undo normalization to 30-day months
+            self._run = self._inputs.loc[n]
+            
+            # Run regression
+            fitted_model = self.run_regression(n)
             
-            # Get long-term normalized gross energy by applying regression result to long-term monthly wind speeds                                                                    
-            gross_norm_por = self._reanalysis_por_avg[self._mc_reanalysis_product[n]].multiply(slope) + intercept
-            gross_por=gross_norm_por*self.num_days_lt/30 # Undo normalization to 30-day months 
+            # Get long-term regression inputs
+            reg_inputs_lt = self.sample_long_term_reanalysis(self._run.num_years_windiness,
+                                                             self._run.reanalysis_product)
 
+            # Get long-term normalized gross energy by applying regression result to long-term monthly wind speeds
+            inputs = np.array(reg_inputs_lt)
+            if num_vars == 1:
+                inputs = inputs.reshape(-1,1)
+            gross_lt = fitted_model.predict(inputs)
```

Comment:
Very good point, Eric. I think that in principle you are right in the sense that ML models do not require that "limitation". On the other hand, I am not sure if this would somewhat conflict with the standard "procedures" currently followed in the sector. Any thoughts, @moptis ?

Reply:
I think Eric is right:  The model should be applied to the full 20-year time series of data, not to any time-averaged data.

Reply:
Cool. Should I change this for the ML models only, or for linear regression too?

Reply:
Maybe for consistency change it for linear regression too. But as Eric says, it won't change the outcome.

Reply:
After a preliminary analysis, we've found that the current method for the long-term correction leads to the most consistent results, so we'll stick with it for now. We'll keep exploring this issue and eventually update the class.